mat/gsvd.go

```diff
@@ -271,68 +275,76 @@ func (gsvd *GSVD) ZeroRTo(dst *Dense) *Dense {
 		dst.Slice(r, k+l, c+r-k-l, c).(*Dense).
 			Copy(b.Slice(r-k, l, c+r-k-l, c))
 	}
-	return dst
 }
 
 // SigmaATo extracts the matrix Σ₁ from the singular value decomposition, storing
 // the result in-place into dst. Σ₁ is size r×(k+l).
-// If dst is nil, a new matrix is allocated. The resulting SigmaA matrix is returned.
 //
-// SigmaATo will panic if the receiver does not contain a successful factorization.
-func (gsvd *GSVD) SigmaATo(dst *Dense) *Dense {
+// If dst is empty, SigmaATo will resize dst to be r×(k+l). If dst is
+// non-empty, SigmATo will panic if dst is not r×(k+l). SigmaATo will also
```

Comment:
"When dst is non-empty, SigmATo will panic if dst is not r×(k+l)."

Reply:
Done.